lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -697,28 +728,42 @@ static Operation *generateFMAOp(vector::FMAOp fmaOp, AIEOpAttributes &opAttr,
                << fmaOp << " into accumulator");
   }
   // If the lhs operand vector is not >= twice the rhs operand vector, then use
-  // concat operator.
+  // concat operator. If targeting for AIE-ML intrinsics, use broadcast operator
+  // for rhs.
   Value lhs = fmaOp.getLhs();
+  Value rhs = fmaOp.getRhs();
   if (!isSimpleVectIntrinsic(fmaOp, state)) {
     AIEVecAttributes lstat = getOperandVecStats(fmaOp, state, 0);
     assert(lstat.vecSizeInBits % 256 == 0);
-    if (lstat.vecSizeInBits == 256) {
-      VectorType concatType =
-          createVectorType(512 / lstat.elementSizeInBits, lstat.elementType);
-      SmallVector<Value> sources = {lhs, lhs};
-      lhs = generateConcatOp(sources, state, fmaOp->getLoc(), concatType);
+    if (AIEML) {
+      rhs = generateBroadcastOp(rhs, stoi(opAttr.start[1]), state,
```

Comment:
Do we need additional checking here to determine that "broadcast" is the right semantics?
The condition !isSimpleVecIntrinsic does not imply that it must be a broadcast.
(see definition of isSimpleVecIntrinsic())

Reply:
Yes, this is on my cleanup list. I will add canGenerateBroadcast() before generateBroadcastOp(). We need to check whether the zbuff is a splat and zoffset is 0x00000000, and xbuffer's accIncr is 1. This will be done after I finish i8 type test, because we might need to add more checks.